toolchain/parse/testdata/basics/fail_bracket_recovery.carbon

```diff
@@ -10,7 +10,11 @@
 
 // This is a valid parse tree even though the lex errors.
```

Comment:
Looks like this is maybe not testing what it used to / intended to test? Maybe something like this would:
```
fn F() {
  (arr[3).F();
}
```
I think the point here is is to create a test where, after bracket recovery inserts a missing `]` before the `)`, the parse then succeeds.

Reply:
Ah yes, I missed this, thanks.
```
  var a: array(i32, 3) = (0, 1, 2);
  // CHECK:STDERR: fail_bracket_recovery.carbon:[[@LINE+4]]:5: error: opening symbol without a corresponding closing symbol [UnmatchedOpening]
  // CHECK:STDERR:   (a[1) + 1;
  // CHECK:STDERR:     ^
  // CHECK:STDERR:
  (a[1) + 1;
```
This looks to test what we want, we see a parse tree with the closing `]` in it:
```
             {kind: 'ParenExprStart', text: '('},
                 {kind: 'IdentifierNameExpr', text: 'a'},
               {kind: 'IndexExprStart', text: '[', subtree_size: 2},
               {kind: 'IntLiteral', text: '1'},
             {kind: 'IndexExpr', text: ']', subtree_size: 4},
           {kind: 'ParenExpr', text: ')', subtree_size: 6},
```